src/test/java/seedu/address/model/loan/NricTest.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.model.loan;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import seedu.address.testutil.Assert;
+
+public class NricTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> new Nric(null));
+    }
+
+    @Test
+    public void constructor_invalidNric_throwsIllegalArgumentException() {
+        String invalidNric = "";
+        Assert.assertThrows(IllegalArgumentException.class, () -> new Nric(invalidNric));
+    }
+
+    @Test
+    public void isValidNric() {
+        // null nric
+        Assert.assertThrows(NullPointerException.class, () -> Nric.isValidNric(null));
+
+        // invalid nric
+        assertFalse(Nric.isValidNric("")); // empty string
+        assertFalse(Nric.isValidNric(" ")); // spaces only
+        assertFalse(Nric.isValidNric("^")); // only non-alphanumeric characters
+        assertFalse(Nric.isValidNric("")); // contains non-alphanumeric characters
```

Comment:
This is the same test case as line 28, and does not have what the comment says it does.